### PR TITLE
APS-66 Add Placement Request Report

### PR DESCRIPTION
**Pull request by davidatkinsuk**
Adding a new 'Placement Application' report and API, to be labelled as 'Placement Request' report on the UI

To facilitate this report i've had to backfill some domain_event 'booking_id' columns, and start capturing the relationship between placement_application_dates and their corresponding placement_requests, also backfilled

An example report is attached (generated from the integration test, so data may look a little 'wonky') - 
[example.xlsx](https://github.com/ministryofjustice/hmpps-approved-premises-api/files/13705655/example.xlsx)

### CODE REVIEW

**Review by pezholio — COMMENTED** [review]
Looking great in general, just a couple of smallish things 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/PlacementRequestService.kt [hunk 1]
```diff
@@ -215,7 +215,12 @@ class PlacementRequestService(
         duration = it.duration,
       )
       val isParole = placementApplicationEntity.placementType == PlacementType.RELEASE_FOLLOWING_DECISION
-      this.createPlacementRequest(placementRequirements, placementDates, notes, isParole, placementApplicationEntity)
+      val placementRequest = this.createPlacementRequest(placementRequirements, placementDates, notes, isParole, placementApplicationEntity)
+
+      it.placementRequest = placementRequest
+      placementDateRepository.save(it)
```
[the remark this answers is not in the document]
Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/PlacementRequestService.kt]: Love it, that is so much neater. Change made.

File: src/main/resources/db/migration/all/20231215134240__create_placement_date_to_request_fk.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE placement_application_dates ADD COLUMN placement_request_id uuid NULL;
+
+ALTER TABLE placement_application_dates ADD CONSTRAINT placement_application_dates_placement_request_id_fkey FOREIGN KEY (placement_request_id) REFERENCES placement_requests(id) DEFERRABLE;
+
+UPDATE placement_application_dates pa_dates SET placement_request_id = (SELECT pr.id FROM placement_requests pr WHERE pr.placement_application_id = pa_dates.placement_application_id AND pr.expected_arrival = pa_dates.expected_arrival)
```
[the remark this answers is not in the document]
Comment (davidatkinsuk) [reply src/main/resources/db/migration/all/20231215134240__create_placement_date_to_request_fk.sql]: Good idea, done

**Review by pezholio — APPROVED** [review]
LGTM - ta! 👍

**Pull request merged**